packages/@storefront/flux-capacitor/src/core/actions/validators.ts

```diff
@@ -181,3 +193,40 @@ export const isNotFetching: Validator<boolean> = {
     !Selectors.infiniteScroll(state).isFetchingBackward,
   msg: 'is already fetching'
 };
+
+export const hasValidLabels: Validator<Actions.Payload.Sort> = {
+  func: ({ labels }) => {
+    return !labels
+      || (
+        Array.isArray(labels)
+        && labels.every((label) => typeof label === 'string')
+      );
+  },
+  msg: 'if present, labels must be an array of strings',
+};
+
+export const hasValidOptions: Validator<Actions.Payload.Sort> = {
+  func: ({ options }) => {
+    return (
+      Array.isArray(options)
+      && !!options.length
+      && options.every((option) =>
+        typeof option.field === 'string'
+        && (!('descending' in option) || typeof option.descending === 'boolean')
+      )
+    );
+  },
+  msg: 'must be an array of valid sort options',
+};
+
+export const hasValidSelected: Validator<Actions.Payload.Sort> = {
+  func: (payload) => {
+    return !('selected' in payload)
+      || (
+        typeof payload.selected === 'number'
+        && payload.selected >= 0
+        && payload.selected <= payload.options.length - 1
+      );
+  },
+  msg: 'if present, must be a number between 0 and `options.length - 1`',
```

Comment:
I would say "is an index of the options array". If you want, you can give the range from 0 to `options.length - 1`, but use the actual calculated value.

Reply:
Done, resolving feedback.